### PR TITLE
REUSE_DB=migrate

### PR DESCRIPTION
**Pull request by snopoke**
@millerdev 

buddy @biyeun

### CODE REVIEW

File: corehq/tests/nose.py [hunk 1]
```diff
@@ -209,26 +213,30 @@ def setup(self):
...
-            for connection in connections.all():
-                db = connection.settings_dict
-                assert db["NAME"].startswith(TEST_DATABASE_PREFIX), db["NAME"]
-                try:
-                    connection.ensure_connection()
-                except OperationalError:
-                    break  # cannot connect; resume normal setup
-                old_names.append((connection, db["NAME"], True))
-            else:
-                self.old_names = old_names, []
```
Comment (millerdev) [inline corehq/tests/nose.py]: Looks like you removed this bit of logic, which is used in [django-nose db teardown](https://github.com/millerdev/django-nose/blob/fast-first/django_nose/plugin.py#L434) and is important for `REUSE_DB=teardown`. I should have documented that in a comment.

File: corehq/tests/nose.py [hunk 2]
```diff
@@ -209,26 +213,30 @@ def setup(self):
...
-                return  # skip remaining setup
+        if self.skip_setup_for_reuse_db and self._databases_ok():
+            if self.run_migrations_for_reuse_db:
+                call_command(
+                    'migrate_multi',
+                    interactive=False,
+                )
```
Comment (millerdev) [inline corehq/tests/nose.py]: nit: could be a single line `call_command('migrate_multi', interactive=False)`

File: corehq/tests/nose.py [hunk 3]
```diff
@@ -177,12 +178,15 @@ class HqdbContext(DatabaseContext):
       rebuilt.
     - `REUSE_DB=teardown` : skip database setup; do normal teardown after
       running tests.
+    - `REUSE_DB=migrate` : skip normal database setup except for running
+      migrations.
```
Comment (millerdev) [inline corehq/tests/nose.py]: This is ambiguous about teardown. What do you think of "same as \`REUSE_DB=1\` except migrate databases before running tests."?

**Comment by czue** [comment]
this is great

File: corehq/tests/nose.py (hunk 1 above)
Comment (snopoke) [inline corehq/tests/nose.py]: restored here: c9a357e

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**